include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -336,14 +336,18 @@ create_geometric_coarsening_sequence(Triangulation<dim, spacedim> const & fine_t
     parallel::distributed::Triangulation<dim, spacedim> temp_tria(
       fine_triangulation->get_communicator(), fine_triangulation->get_mesh_smoothing());
 
+    MPI_Comm mpi_comm = fine_triangulation_in.get_communicator();
```

Comment:
why don't you work with the communicator of `fine_triangulation`? In my opinion, it would be more natural to work with the tria casted to `p::d::T` at this point of the code, e.g. to make the flow of information more clear or to allow to split this function into smaller sub-units.